cmd/kube-apiserver/app/options/options.go

```diff
@@ -168,7 +168,11 @@ func (s *APIServer) AddFlags(fs *pflag.FlagSet) {
 	fs.BoolVar(&s.ServiceAccountLookup, "service-account-lookup", s.ServiceAccountLookup, "If true, validate ServiceAccount tokens exist in etcd as part of authentication.")
 	fs.StringVar(&s.KeystoneURL, "experimental-keystone-url", s.KeystoneURL, "If passed, activates the keystone authentication plugin")
 	fs.StringVar(&s.AuthorizationMode, "authorization-mode", s.AuthorizationMode, "Ordered list of plug-ins to do authorization on secure port. Comma-delimited list of: "+strings.Join(apiserver.AuthorizationModeChoices, ","))
-	fs.StringVar(&s.AuthorizationPolicyFile, "authorization-policy-file", s.AuthorizationPolicyFile, "File with authorization policy in csv format, used with --authorization-mode=ABAC, on the secure port.")
+	fs.StringVar(&s.AuthorizationConfig.PolicyFile, "authorization-policy-file", s.AuthorizationConfig.PolicyFile, "File with authorization policy in csv format, used with --authorization-mode=ABAC, on the secure port.")
+	fs.StringVar(&s.AuthorizationConfig.WebhookURL, "authorization-webhook-url", s.AuthorizationConfig.WebhookURL, "URL of a remote service to recieve a webhook request, used with --authorization-mode=Webhook, to defer to authorization on the secure port.")
+	fs.StringVar(&s.AuthorizationConfig.WebhookCertFile, "authorization-webhook-cert-file", s.AuthorizationConfig.WebhookCertFile, "If set, the file containing x509 Certificate the webhook HTTP client will use to perform HTTPS client authentication.")
+	fs.StringVar(&s.AuthorizationConfig.WebhookKeyFile, "authorization-webhook-key-file", s.AuthorizationConfig.WebhookKeyFile, "File containing x509 private key matching --authorization-webhook-cert-file.")
+	fs.StringVar(&s.AuthorizationConfig.WebhookCAFile, "authorization-webhook-ca-file", s.AuthorizationConfig.WebhookCAFile, "If set, the remote service's certificate will be verified by one of the authorities in the file, otherwise the host's root CA set will be used.")
```

Comment:
This is a lot of new flags. Since you're using client certs, and since our kubeconfig files are capable of storing client certs, what do you think about just requiring a kubeconfig file configured for the webhook destination?